src/main/java/com/aws/greengrass/mqtt/bridge/MessageBridge.java

```diff
@@ -146,8 +155,29 @@ private <T extends Message> void handleMessage(TopicMapping.TopicType sourceType
     private <T extends Message> void publishMessage(MessageClient<T> client, String topic, Message message)
             throws MessageClientException {
         T msg = client.convertMessage(message);
-        msg = (T) msg.newFromMessageWithTopic(topic);
-        client.publish(msg);
+
+        if (!isRetainAsPublished(topic) && msg instanceof MqttMessage) {
+            MqttMessage mqttMsg = (MqttMessage) msg;
+            mqttMsg = mqttMsg.toBuilder().topic(topic).retain(false).build();
+            client.publish((T) mqttMsg);
+        } else {
+            msg = (T) msg.newFromMessageWithTopic(topic);
+            client.publish(msg);
+        }
+    }
+
+    private boolean isRetainAsPublished(String topic) {
+        if (optionsByTopic.isEmpty()) {
+            return DEFAULT_RETAIN_AS_PUBLISHED;
+        }
+        Map<String, Mqtt5RouteOptions> opts = new HashMap<>();
+        for (Map.Entry<String, TopicMapping.MappingEntry> entry : topicMapping.getMapping().entrySet()) {
+            if (topic.equals(entry.getValue().getTopic())) {
+                String route = entry.getKey();
+                opts.put(entry.getValue().getTopic(), optionsByTopic.get(route));
+            }
```

Comment:
this is going to do a loop on every single publish. We really can't do that.

Reply:
Makes sense, I'll try to find a way where we can pass in the optionsByTopic in the right format so we don't have to build it out each time

Reply:
I edited it to pass in the optionsByTopic in the expected format when the message bridge is created at install

Reply:
Does that correctly handle updates when subscriptions and configurations change?

Reply:
Hm as it stands it might not, I think I'll need to add the mqtt5 route options to the list of config changes that would require a reinstall